others/connect_four_game.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+* @file
+* @brief [Connect Four](https://en.wikipedia.org/wiki/Connect_Four)
+*
+* @details
+* This is a user interactive Connect-Four game for two players.
+* A board with 6 rows and 8 columns is displayed.
+* Two players are assigned the symbols "x" and "o".
+* They take turns to choose the column where their respective symbol can be inserted.
+* A player wins when 4 spaces are connected continuously: Horizontally, Vertically, Diagonally Right, or Diagonally Left.
+* The program declares the winner.
+*
+* @author [Arunabh Bhattacharya](https://github.com/arunabh-cloud)
+*/
+#include <iostream>   /// for IO operations
+#include <array>     /// for std::array
+using std::cout;
+using std::cin;
+
+const int COLS = 8;  /// The number of board columns is 8
+const int ROWS = 6;  /// The number of the board rows is 6
+
+/**
+  * @details 
+  * Method for printing the board with tokens or '-' if a space is yet to be chosen
+  * @param board of dimension ROWS * COLS
+  * @param ROWS is the vertical dimension of the board
+  * @param COLS is the horizontal dimension of the board
+  */
+void printBoard(std::array<std::array<char, COLS>, ROWS>&board) {
+    for (int i = ROWS - 1; i > -1; i--) {
+        for (int j = 0; j < COLS; j++) {
+            std::cout << board[i][j] << " ";
+        }
+        std::cout << std::endl;
+    }
+}
+
+/**
+  * @details
+  * Method for initializing and printing the board with "-"
+  * @param board of dimension ROWS * COLS
+  * @param ROWS is the vertical dimension of the board
+  * @param COLS is the horizontal dimension of the board
+  */
+void initializeBoard(std::array<std::array<char, COLS>, ROWS>&board) {
+    for (int i = 0; i < ROWS; i++) {
+        for (int j = 0; j < COLS; j++) {
+            board[i][j] = '-';
+        }
+    }
+    printBoard(board);
+    std::cout << std::endl;
+    std::cout << "Player 1: x\n";  ///  Assign the token 'x' to the Player 1 
+    std::cout << "Player 2: o\n";  /// Assign the token 'o' to the Player 2 
+    std::cout << std::endl;
+}
+
+/**
+  * Play the game till there is a winner or till the game draws
+  * @param board of dimension ROWS * COLS
+  * @param ROWS is the vertical dimension of the board
+  * @param COLS is the horizontal dimension of the board
+  */
+void playGame(std::array<std::array<char, COLS>, ROWS>&board){
+    /// Initializing game variables
+    int count = 1;     /// Counts each turn played till the board is full
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂